linkedList.go

```diff
@@ -89,37 +108,127 @@ func (l *L) Remove(index uint) bool {
+		Index uint `json:"index"`
+		Value int  `json:"value"`
+	}
+
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, "Invalid request payload", http.StatusBadRequest)
+		return
+	}
+
+	success := list.Insert(req.Index, req.Value)
+	if !success {
+		http.Error(w, "Insert failed", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(map[string]string{"message": "Insert successful"})
+}
+
+func handleGet(w http.ResponseWriter, r *http.Request, list *LinkedList) {
+	indexStr := strings.TrimPrefix(r.URL.Path, "/get/")
+	index, err := strconv.Atoi(indexStr)
+	if err != nil {
+		http.Error(w, "Invalid index", http.StatusBadRequest)
+		return
+	}
+
+	value, found := list.Get(uint(index))
+	if !found {
+		http.Error(w, "Index out of range", http.StatusNotFound)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]int{"value": value})
+}
+
+func handleRemove(w http.ResponseWriter, r *http.Request, list *LinkedList) {
+	indexStr := strings.TrimPrefix(r.URL.Path, "/remove/")
+	index, err := strconv.Atoi(indexStr)
+	if err != nil {
+		http.Error(w, "Invalid index", http.StatusBadRequest)
+		return
+	}
+
+	success := list.Remove(uint(index))
+	if !success {
+		http.Error(w, "Remove failed", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]string{"message": "Remove successful"})
+}
 
-	value, ok := newList.Get(1) // value at index 1
-	if !ok {
-		fmt.Println("failed, index out of range")
-	} else {
-		fmt.Println("value at index 1:", value)
+func handleFind(w http.ResponseWriter, r *http.Request, list *LinkedList) {
+	valueStr := strings.TrimPrefix(r.URL.Path, "/find/")
+	value, err := strconv.Atoi(valueStr)
+	if err != nil {
+		http.Error(w, "Invalid value", http.StatusBadRequest)
+		return
 	}
 
-	value, ok = newList.Get(8) // value at an out-of-range index
-	if !ok {
-		fmt.Println("Retrieval failed, index out of range")
-	} else {
-		fmt.Println("Value at index 8:", value)
+	index, found := list.Find(value)
+	if !found {
+		http.Error(w, "Value not found", http.StatusNotFound)
+		return
 	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(map[string]uint{"index": index})
+}
+
+func handleList(w http.ResponseWriter, _ *http.Request, list *LinkedList) {
+	current := list.head
+	var values []int
+	for current != nil {
+		values = append(values, current.Value)
+		current = current.Next
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(values)
+}
+
+func main() {
+	list := NewLinkedList()
+	http.HandleFunc("/insert", func(w http.ResponseWriter, r *http.Request) {
+		handleInsert(w, r, list)
+	})
+	http.HandleFunc("/get/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
It is a good idea to specify the HTTP Method as well.
For example this handler function will be called for all the HTTP Method. but it should only be used for GET Method.